engines/osxnative/engine.go

```diff
@@ -0,0 +1,51 @@
+//go:generate go-composite-schema --unexported --required engine payload-schema.yml generated_payloadschema.go
+
+// Package osxnative implements the Mac OSX engine
+package osxnative
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type engine struct {
+	engines.EngineBase
+	log *logrus.Entry
+}
+
+type engineProvider struct {
+	extpoints.EngineProviderBase
+}
+
+func init() {
+	// Register the mac engine as an import side-effect
+	extpoints.EngineProviders.Register(engineProvider{}, "macosx")
+}
+
+func (e engineProvider) NewEngine(options extpoints.EngineOptions) (engines.Engine, error) {
+	return engine{log: options.Log}, nil
+}
+
+func (e engineProvider) ConfigSchema() runtime.CompositeSchema {
+	return runtime.NewEmptyCompositeSchema()
+}
+
+func (e engine) PayloadSchema() runtime.CompositeSchema {
+	return payloadSchema
+}
+
+func (e engine) NewSandboxBuilder(options engines.SandboxOptions) (engines.SandboxBuilder, error) {
+	taskPayload := options.Payload.(*payload)
```

Comment:
if we do a type conversion here, and there is an error, this will cause the worker to panic and quit.

What we could possibly do is:
```
taskPayload, ok := options.Payload.(*payload)
if !ok {
  <do something here to log that there was an error and handle gracefully>
}
<continue on here if everything is ok>
```

Here is an example: https://play.golang.org/p/0EYrxyjhgC